File: valid parenthesis/ValidParenthesis.java
class ValidParenthesis {
    public boolean isValid(String s) {
        int test = 0;
        Deque<Integer> deq = new LinkedList<>();
        try{
            for(int i = 0; i < s.length(); i++){
                switch(s.charAt(i)){
                    case '(':
                       deq.push(1);
                        break;
                    case '{':
                        deq.push(2);
                        break;
                    case '[':
                        deq.push(3);
                        break;
                    case ')':
                        test = deq.pop();
                        if(test != 1) return false;
                        break;
                    case '}':
                        test = deq.pop();
                        if(test != 2) return false;
                        break;
                    case ']':
                        test = deq.pop();
                        if(test != 3) return false;
                }
            }
            return deq.poll() == null;
        }
        catch (Exception e){return false;}
    }
}

File: valid parenthesis/valid_parenthesis.py
class Solution:
    def isValid(self, s: str) -> bool:
        x = []
        try:
            for c in s:
                if c == '(':
                    x.append(1)
                if c == '{':
                    x.append(2)
                if c == '[':
                    x.append(3)
                if c == ')':
                    if x.pop() != 1:
                        return False
                if c == '}':
                    if x.pop() != 2:
                        return False
                if c == ']':
                    if x.pop() != 3:
                        return False
            return len(x) == 0
        except Exception:
            return False
            
        